Add tooltips to each balance title in the account balances card

Jinksi opened the pull request:
Resolves #5724

### Changes proposed in this Pull Request

- Adds a help icon tooltip to each of the account balance card fund titles.
- When a help icon tooltip is clicked, a tooltip will be rendered.
- ~Updates the `Tooltip` component with an `ignoreMouseHover` props~  
Uses the ClickTooltip component to show the tooltip content only via click interactions.
- Adds documentation URLs to the Account Balances `strings.ts` file.
- Reduces the coupling between BalanceBlock and content, which is now passed down as props (https://github.com/Automattic/woocommerce-payments/pull/5805#discussion_r1145376190).

> **Note**
> ~I am concerned about the~ (**Updated in b1c1b3e3971659c3023cb0b7b74e24f369e433cc**) low contrast of links rendered as tooltip content. I think this should be raised when we request design feedback, since this is a WCPay-wide component.

https://user-images.githubusercontent.com/3147296/230227968-0c846fc1-f363-421c-b703-095de26656c6.mov

**Available funds**
_"The amount of funds available to be deposited. Learn more."_
- 🔗 https://woocommerce.com/document/woocommerce-payments/deposits/deposit-schedule/ 

![8453d6dbe03b1ced521b13531bf93a26](https://user-images.githubusercontent.com/3147296/230228074-c6c42f06-c651-4c25-9566-615f06b40e69.png)


**Negative available funds**
- _"Learn more about why your account balance may be negative."_
- 🔗 https://woocommerce.com/document/woocommerce-payments/fees-and-debits/account-showing-negative-balance

![75004307573fccbd0d491f406cb2a54a](https://user-images.githubusercontent.com/3147296/230228052-9cec3039-e708-46d8-ab9a-0ea629be9461.png)


**Pending funds**
- _"The amount of funds still in the X day pending period. Learn more."_
- ❗ Note: The X will need to vary based on the merchant country and their pending period. 
- 🔗 https://woocommerce.com/document/woocommerce-payments/deposits/deposit-schedule/

![5b6aa9d183a37b03246ab091c2f23256](https://user-images.githubusercontent.com/3147296/230228024-6b129e6f-7009-4414-98a9-ddbdb1e57a03.png)

### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Enable the `_wcpay_feature_simplify_deposits_ui` feature flag.
  To enable, run `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
* Navigate to Payments → Overview.
* View each tooltip by clicking it. Hovering should not trigger the tooltip to be displayed.
* Ensure the correct text and documentation link is rendered, as listed above.
* To test the negative account balance tooltip, make a test purchase with a disputed test card number (`4000000000000259`).

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Jinksi committed:
Refactor Tooltip and TooltipBase components to TS

Jinksi committed:
Refactor `Tooltip` component unit test files to TS

Added the `React` import to fix eslint errors

Jinksi committed:
Add changelog entry

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Add final return statement to ensure boolean return type

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Add balance tooltip strings

Jinksi committed:
Add BalanceTooltip component to BalanceBlock

Jinksi committed:
Add type for `<TooltipPortal />`

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Update JSDoc comments

Jinksi committed:
Add prop to `Tooltip` component for ignoring hover events

Jinksi committed:
Ensure BalanceTooltip ignores hover events

Jinksi committed:
Update AccountBalances snapshot test

Jinksi commented on client/components/tooltip/test/index.test.tsx:
```diff
@@ -82,6 +88,111 @@ describe( 'Tooltip', () => {
+		expect( handleHideMock ).toHaveBeenCalled();
+	} );
+
+	it( 'renders and hides its content when hovered', () => {
```
Added a test to ensure show/hide on hover events.

Jinksi commented on client/components/tooltip/test/index.test.tsx:
```diff
@@ -82,6 +88,111 @@ describe( 'Tooltip', () => {
+		expect( handleHideMock ).toHaveBeenCalled();
+	} );
+
+	it( 'remains hidden when hovered ignoreMouseHover = true', () => {
```
Added a test to ensure NO show/hide on hover events when ignoreMouseHover = true.

Jinksi commented on client/components/tooltip/test/index.test.tsx:
```diff
@@ -82,6 +88,111 @@ describe( 'Tooltip', () => {
 		expect( handleHideMock ).toHaveBeenCalled();
 	} );
 
+	it( 'renders and hides its content when clicked and ignoreMouseHover = true', () => {
```
Added a test to ensure show/hide on click events when ignoreMouseHover = true.

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Add N/A changelog file

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5805` or branch name `add/5724-account-balances-tooltips` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5724-account-balances-tooltips) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7686883258a0a1b81bf68269db63abcc8c9c0d1f
- Build time: 2023-04-06 01:08:31 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Add accessible role and label to Tooltip button element

Jinksi committed:
Add tests for BalanceTooltip component

Jinksi committed:
Adjust icon color, alignment and margin to match design

Jinksi commented on client/components/account-balances/balance-tooltip.tsx:
```diff
@@ -0,0 +1,106 @@
+	let content = null;
+	const tooltipButtonLabel = `${ fundLabelStrings[ type ] } tooltip`;
+
+	if ( type === 'available' ) {
```
Creating an `if` statement for each type is verbose, but I kept it like this for clarity.

Jinksi committed:
Update AccountBalances snapshot test

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

shendy-a8c commented:
Before reviewing too deep, I feel like the `BalanceTooltip` knows too much of the business logic, needing the `delayDays` to 'pierce' through the `BalanceBlock` even though only 'Pending fund' that requires that parameter.

How about making the tooltip component 'dumber' and pass the tooltip content from `AccountBalancesTabPanel`? What do you think?

haszari commented on client/components/account-balances/balance-tooltip.tsx:
```diff
@@ -0,0 +1,107 @@
+				</a>
+			</>
+		);
+	}
```
Agree with @shendy-a8c comment – can we make the tooltip component dumb and just pass in the content that it should display?

Jinksi replied:
Thanks, @shendy-a8c, @haszari this makes sense. I'll rework this and see if I can simplify it.

Jinksi replied:
I've changed this up in 1649288fbee7bb9754f2da54d72efccdfabc5047.

I moved the logic for determining the tooltip content up into the `AccountBalancesTabPanel` component, which reduces the amount of prop drilling required.

Let me know what you think 🙏 

I could move the `type` prop out too, but that would start to introduce duplicate logic.

haszari replied:
Is `type` for behaviour or content? It looks like it's used as a [constant / array index](https://github.com/Automattic/woocommerce-payments/blob/8d62900a1137f065a562a38f54c72d361c1b7f3a/client/components/account-balances/balance-block.tsx#L63) AND [rendered in the UI as a tooltip](https://github.com/Automattic/woocommerce-payments/blob/8d62900a1137f065a562a38f54c72d361c1b7f3a/client/components/account-balances/balance-block.tsx#L56).

I'd suggest passing the content in direct – e.g. pass the title and the tooltip as strings. As it stands the `BalanceBlock` component is coupled to `fundLabelStrings`; IMO that coupling would be better higher up.

Jinksi replied:
Thanks @haszari. I've updated BalanceBlock and BalanceBlockTooltip to accept content from the TabPanel component in cf4908d0f7a83d0f7c9d9b61f47a32ce8e2f224c.

`type` is now being used only for generating a unique ID  for each type & currency to be used by aria-labelledby.

haszari replied:
Sounds like type is now `labelledby` or `id` or `accessibilityparent` or something – maybe that param can be 'dumber' :)

Jinksi replied:
Yeh, it can get dumber for sure. I'll change the label as well.

The tradeoff is that we start to duplicate lines of code (e.g. `wcpay-account-balances-${ currencyCode }-${ type }-title`) which could introduce a risk of human error. However, this is not a huge problem for 2-3 elements.

Jinksi replied:
Updated in 3ec9aacfcd99814026a0d3ea6d8fdffa7c7327e1.

haszari replied:
Gotcha, I can see why a generic `balancelabel` or similar might be a better fit.

haszari commented on client/components/tooltip/index.tsx:
```diff
@@ -11,23 +11,31 @@ import TooltipBase, { TooltipBaseProps } from './tooltip-base';
 
 const Tooltip: React.FC< TooltipProps > = ( {
 	isVisible,
+	ignoreMouseHover = false,
```
Can we make this parameter more semantic and/or positively worded? The "display on click, hide on click elsewhere" behaviour is a new mode, necessary for tooltips that contain links to docs. What's a good name for that mode? Or can we infer that mode from the content passed in?

Jinksi replied:
> Can we make this parameter more semantic and/or positively worded?

`clickEventsOnly`? Another option is to make a separate component, e.g. `ClickableTooltip`. What do you think?

> Or can we infer that mode from the content passed in?

I think being more explicit about the behaviour is preferred. There are uses of this Tooltip component where a "Learn more" doc link exists within its content, e.g the settings screen:
<img width="321" alt="image" src="https://user-images.githubusercontent.com/3147296/227041918-dbda1c5a-5b88-459f-b4df-98b36b3f53b0.png">

> necessary for tooltips that contain links to docs.

If this "clickable" tooltip is to be required for all tooltips with docs links, I could move this to a new issue/PR to update the component and change the existing behaviour.

haszari replied:
> If this "clickable" tooltip is to be required for all tooltips with docs links, I could move this to a new issue/PR to update the component and change the existing behaviour.

Pretty sure that's the case (my assumption – might be clarification in design comments in figma or P2). You could confirm with @elizaan36 , here's how I understand it:

- Existing help icon + tooltip component shows a text message on mouseover/hover. The message auto-hides (i.e. mouse away / timeout).
- The ones we need look similar but behave differently, because the user needs to be able to click on the docs link. This behaviour should be used for any help icon+tooltip that has a link (or maybe for long messages, etc).

haszari replied:
> I think being more explicit about the behaviour is preferred.

I agree :) 

> ClickableTooltip

Sounds reasonable :)

elizaan36 replied:
+1 on everything here. This is a new variant/behavior for the existing icon + tooltip to account for the docs link. It's also optimized for mobile.

Jinksi replied:
Thanks @elizaan36 and @haszari, I'll make it happen 👍

Jinksi replied:
I've created and assigned myself to a new issue #5851 for implementing the new click-only Tooltip variant and updating existing tooltips with doc links. I'll update this PR description to mention this.

Jinksi committed:
Pass the BalanceTooltip as a prop to simplify required props

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into dev/ts-refactor-tooltip-component

Jinksi committed:
Merge branch 'dev/ts-refactor-tooltip-component' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Split Tooltip into HoverTooltip and ClickTooltip

Jinksi committed:
Refactor tooltip tests to include new components

Jinksi committed:
Show pointer cursor for ClickTooltip variant

Jinksi committed:
Merge branch 'develop' into add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux

Jinksi committed:
Add changelog entry

Jinksi committed:
Use ClickTooltip for payment method checkboxes to improve mobile UX

Jinksi committed:
Merge branch 'add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux' into add/5724-account-balances-tooltips

Jinksi committed:
Remove changes to Tooltip components after changing base

Jinksi committed:
Update test snapshot

Jinksi committed:
Move tooltip content definition out of BalanceTooltip

to reduce the amount of prop drilling required

Jinksi committed:
Update BalanceBlockProps JSDoc

Jinksi committed:
Raise BalanceBlock props to reduce prop drilling

Jinksi committed:
Remove unnecessary export for balanceType

Jinksi committed:
Pass BalanceBlock `id` as prop to somplify component

james-allan committed:
Merge branch 'develop' into add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux

Jinksi committed:
Merge branch 'develop' into add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux

Jinksi committed:
Revert instances of ClickTooltip to HoverTooltip

As part of this PR, existing Tooltip components should become HoverTooltip.

Jinksi committed:
Merge branch 'add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux' into add/5724-account-balances-tooltips

Jinksi committed:
Remove reserved tooltip strings

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Revert inadvertent change after merge

Jinksi committed:
Fix incorrect available/negative balance tooltip

Jinksi commented:
Now that #5862 has been merged, this PR is ready for review.

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

shendy-a8c committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

shendy-a8c commented on client/components/account-balances/balances-tab-panel.tsx:
```diff
@@ -71,16 +83,73 @@ const AccountBalancesTabPanel: React.FC = () => {
+								label={ `${ fundLabelStrings.available } tooltip` }
+								content={
+									tab.availableFunds < 0 ? (
+										<a
```
Can we use `ExternalLink` from `@wordpress/components` here?

There is UI difference with design. It has the arrow icon.

![Screenshot 2023-04-03 at 00 19 31](https://user-images.githubusercontent.com/73803630/229368573-89c9abd2-1090-454c-a3fa-1c2620763bd9.png)

It will open the link not in the same page without specifying target="_blank" and it's used in several places, so maybe that's the standard.

I didn't think of it before but now I wonder what is considered 'external'. Is our own docs external? Or maybe it means every link that does not point to admin pages is external?

elizaan36 replied:
The external icon indicates that it will open a new tab. I think there's a bug with the WordPress component where the icon surfaces an unwanted string of text. Will you check that before adding?

Jinksi replied:
Thanks for the heads up @elizaan36.

I've used ExternalLink and made sure the `(opens in new tab)` string does not display in ca83eac331b1d1e071a3f56d630bdffb831e1c45.

Please see the updated screenshots in the PR description.

shendy-a8c commented on client/components/account-balances/balances-tab-panel.tsx:
```diff
@@ -71,16 +83,73 @@ const AccountBalancesTabPanel: React.FC = () => {
+								label={ `${ fundLabelStrings.pending } tooltip` }
+								content={
+									<>
+										{ sprintf(
```
I have similar situation with [my PR](https://github.com/Automattic/woocommerce-payments/pull/5861/files#r1151093049).

I agree that it's nice to have strings in one place, in `strings.ts`, but here we have dynamic string and we assume that the string has a delay days parameter.

I don't know the ideal solution though. I consider my concern as `nit`, so feel free to ignore.

haszari replied:
I don't think we should factor strings out of the components – it decouples the format string (placeholders / interpolation) from the replacements. 

@Jinksi do you know why we have separate strings files? What do you recommend?

Jinksi replied:
We decided to follow this `strings.ts` convention when we were discussing how we'll setup the skeleton components. It seems to be used throughout WCPay.

There are pros/cons, I don't have a preference either way.

I've moved these strings inside the component file in 10bc97e8a0a2d20518f106dc759fa38e1906c54c; it becomes less cumbersome when dealing with `interpolateComponents`.

haszari replied:
> We decided to follow this strings.ts convention when we were discussing how we'll setup the skeleton components. It seems to be used throughout WCPay.

TIL, good to know. Something for the front-end guild to consider maybe.

shendy-a8c commented on client/components/account-balances/strings.ts:
```diff
@@ -24,5 +24,29 @@ export const fundLabelStrings = {
+		'woocommerce-payments'
+	),
+	pending: __(
+		'The amount of funds still in the %d day pending period.',
```
Do we want to get out of our ways to shows `days` when %d is larger than 1?

shendy-a8c replied:
Also, 
[delay days can be zero](https://stripe.com/docs/api/accounts/object#account_object-settings-payouts-schedule-delay_days). We should handle such case. I'm not sure when it's zero though, maybe when schedule is set to manual?

Jinksi replied:
> Do we want to get out of our ways to shows days when %d is larger than 1?

I think the current grammar is correct for any %d value, e.g.:

* the 1 day pending period
* the 2 day pending period
* the 0 day pending period

> We should handle such case. I'm not sure when it's zero though, maybe when schedule is set to manual?

I'm also not sure if this could occur in reality. We could hide the tooltip, or we could state "the 0 day pending period", which is technically correct.

james-allan replied:
Even though the grammar is correct in english, I think we may still need to use a `_n` here. 

Some languages change the order of the words so that the grammar is no longer correct. eg in French the translation is "la période d'attente de 2 jours" which translates into "The waiting period is 2 days". By keeping this a simple `__()` it may complicate it for some translators to squeeze their language into a non-plural format.

Jinksi replied:
Thanks for mentioning this @james-allan. TIL about `_n()`, I've updated this in 1bfc4a7c2ed9de90bbd588a5fba9c1184792e5e1. Does it look like I'm using it correctly here?

https://github.com/Automattic/woocommerce-payments/blob/1bfc4a7c2ed9de90bbd588a5fba9c1184792e5e1/client/components/account-balances/balances-tab-panel.tsx#L137-L142

james-allan replied:
Yep that's correct. :)

shendy-a8c commented on client/components/account-balances/test/index.test.tsx:
```diff
@@ -290,3 +292,64 @@ describe( 'AccountBalancesTabPanel', () => {
+	test( 'renders the correct tooltip text for the pending balance', () => {
+		const delayDays = 17;
+		// Insert the delayDays value into the expected tooltip text.
+		const expectedTooltipText = fundTooltipStrings.pending.replace(
```
Should we use hardcoded string in the unit test? This is so the unit test 'locks' the code and can detect if there is changes in the code.

With using variable `fundTooltipStrings.pending` here, when it's changed in the future, this unit test will not fail and probably won't aware where are the places that are affected by the change.

What do you think?

haszari replied:
> This is so the unit test 'locks' the code and can detect if there is changes in the code.

As I understand it, that's the pattern – the test should hard-code the expectations and ideally only depend on the component being tested.

Jinksi replied:
Thanks @shendy, @haszari! I've since refactored the tests in 845e8738d4b9179d87c8886b1c8e6df1336b6789 with this in mind, since the previous tests made no sense now that the strings have been moved out of `strings.ts`.

I've kept the documentation URLs as imported variables from `strings.ts` however, as I think these should be kept in a separate file for maintainability purposes.

shendy-a8c reviewed (dismissed):
Overall looks good and tested well. I made sure the strings and links are correct per https://github.com/Automattic/woocommerce-payments/issues/5724.

That issue says:
> Check the design comments in Figma for the latest copy.

Where is that exactly? I assume the strings in the issues are still up to date?

Another thing that I think very minor is that the pointer jitters when hovered on the icon, when the tooltip is open.

https://user-images.githubusercontent.com/73803630/229369728-42e842d4-8838-40d6-a642-ecaad86765d2.mov

shendy-a8c commented:
I notice when in loading state, the tooltip is there already and shows the default string. For example, even if I have negative balance, it will show the text and link for the non-negative balance for the Available funds panel. Once the data is fetched, it will correct itself.

Maybe the tooltip should not show until data is fully loaded?

This is very minor IMO.

elizaan36 commented:
hey all, quick note.

For this tooltip,
<img width="271" alt="image" src="https://user-images.githubusercontent.com/5121465/229371005-e0ab641d-d031-45a3-8e81-721e48c09a15.png">

Only "Learn more" should be linked in this sentence. I'm assuming this is pulling in the link accent color from the user theme but I'm concerned about the color contrast for accessibility. Is there any way to lighten the blue on the black background?

The same link accent color should be pulled in for the question mark icon. This is the indication that they are different to regular tooltips that only show on hover. The link color will provide more of an affordance that they should be clicked.

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Don't obscure mouse target el with tooltip arrow

Jinksi committed:
Use hard-coded strings in tests for clarity

Jinksi committed:
Hide account balance tooltips while loading data

Jinksi committed:
Use ExternalLink component for tooltip links

Jinksi committed:
Move tooltip string definitions into component

Jinksi committed:
Add role to tooltip component to improve a11y

Jinksi committed:
Refactor BalanceTooltip tests to integration tests

Jinksi committed:
Change tooltip icon color to wp-admin theme color

Jinksi commented on client/components/account-balances/style.scss:
```diff
@@ -59,6 +59,39 @@
+	align-items: center;
+	margin-left: 0.5em;
+
+	&__icon {
+		fill: $gray-900;
+	}
+}
```
Using the same color as is used by `@wordpress/components` `Button`

Jinksi commented on client/components/tooltip/style.scss:
```diff
@@ -43,6 +43,8 @@
 			transform: translate( -50%, 22px );
 			border: solid 15px transparent;
 			border-top-color: $gray-900;
+			// Ensure the tooltip arrow does not obscure the mouse target element.
+			pointer-events: none;
 		}
 	}
```
Using a lighter blue color from the `@wordpress/base-styles` library.

https://github.com/WordPress/gutenberg/blob/d0fa04737ffda96b03bade9d724bc7fd842482d4/packages/base-styles/_colors.native.scss#LL33C23-L33C23

Jinksi committed:
Lighten the color of links within tooltips to improve a11y

Jinksi commented on client/components/tooltip/style.scss:
```diff
@@ -43,6 +43,12 @@
+		}
+
+		a {
+			color: $blue-medium;
```
Using a lighter blue color from the `@wordpress/base-styles` library.

https://github.com/WordPress/gutenberg/blob/d0fa04737ffda96b03bade9d724bc7fd842482d4/packages/base-styles/_colors.native.scss#LL33C23-L33C23

Jinksi commented on client/components/account-balances/style.scss:
```diff
@@ -59,6 +59,39 @@
+	margin-left: 0.5em;
+
+	&__icon {
+		fill: var( --wp-admin-theme-color );
```
Using the same color as is used by `@wordpress/components` `Button`

Jinksi commented:
Thanks for you feedback @shendy-a8c.

I've made changes to this PR based on your feedback. I've replied to your comments, but to address your other points:

> > Check the design comments in Figma for the latest copy.
> 
> Where is that exactly? I assume the strings in the issues are still up to date?

I'll attempt a Figma GH shorthand link: q2b4LTEUhyw0y1vKmfEVpx-fi-434-61624.

Look at the "i1"→ "Overview – Full Page – Multi currency" design. The content for each tooltip can be found in the comments above each tooltip.

> Another thing that I think very minor is that the pointer jitters when hovered on the icon, when the tooltip is open.

I've added this in 2b4b11400c460a466f153389bcd311cc2dddff1c.

> Maybe the tooltip should not show until data is fully loaded?

Updated in ea7a0eea70b5397bcbe2165c0339fc7161e3bb93.

Jinksi commented:
> Only "Learn more" should be linked in this sentence.

Thanks for catching this @elizaan36, I've updated this to only wrap "Learn more" as a link element.

> Is there any way to lighten the blue on the black background?

I've opted to use the colour `$blue-medium` (`#00aadc`) from the @wordpress/base-styles library for links within tooltips. Let me know what you think of this choice.

> The same link accent color should be pulled in for the question mark icon.

I've updated this in 59c67544ef37fcd465c70e01babf5b3b899a751d.

I've also updated the screenshots in the PR description with all of these changes.

james-allan commented on client/components/account-balances/balances-tab-panel.tsx:
```diff
@@ -71,16 +79,79 @@ const AccountBalancesTabPanel: React.FC = () => {
+										  } )
+										: interpolateComponents( {
+												mixedString: __(
+													'The amount of funds available to be deposited. {{learnMoreLink}}Learn More.{{/learnMoreLink}}',
```
I haven't been following this review closely so please ignore me if this has been discussed but is this capitalisation correct? Should this be **Learn more** instead of **Learn More**.

Jinksi replied:
Ah! Yeh it is meant to be **'Learn more'**. Now I have to do the screenshots again 🤦‍♂️

Thanks for catching this.

Jinksi replied:
Updated in c2e52f6a6c61f4ae8c5785f2aa2a681f38af0d9f

Jinksi committed:
Allow plural translations for pending tooltip content.

Jinksi committed:
Fix incorrect capitalisation in 'Learn more' strings

james-allan reviewed (approved):
@Jinksi this looks good to me.

I noticed you asked for confirmation about the use of the link color (#00aadc) in your most recent comment. There's a thumbs up from Elizabeth on that so I take that's approval. 

The other comments about checking the latest copy from the designs. That was just a precaution in case they were changed since I included them in the issue. It doesn't look like they have changed though so this is fine. 

I've tested the negative available balance, positive available balance and pending tooltips and they all look inline with the feedback and designs.

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi commented:
As per discussion in https://github.com/Automattic/woocommerce-payments/pull/5861#issuecomment-1497602158, I'll remove the "ExternalLink" icons before merging.

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Remove ExternalLink icons from tooltip content

Jinksi commented:
I've removed the external link icons in a674faa and updated the screen captures in the PR description. @james-allan are you happy for me to merge after these changes?

james-allan reviewed (approved):
Yep. The most recent changes to use `<a>` tags looks good to me and works as intended.

Jinksi commented:
@shendy-a8c since you requested changes and merging is blocked, I'll await your review before merging.

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi committed:
Merge branch 'develop' into add/5724-account-balances-tooltips

Jinksi commented:
Failing E2E test seems unrelated:
```
Shopper > Checkout > Failures with various cards › should throw an error that the card was declined due to invalid 3DS card
```
